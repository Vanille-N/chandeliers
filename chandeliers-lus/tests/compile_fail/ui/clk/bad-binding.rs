chandeliers_lus::decl! {
    extern node random_bool() returns (b : bool);

    node system() returns ();
    var b1, b2 : bool;
        _n : int when b1;
    let
        b1 = random_bool();
        b2 = random_bool();
        _n = 1 when b2;
    tel
}

fn main() {}
