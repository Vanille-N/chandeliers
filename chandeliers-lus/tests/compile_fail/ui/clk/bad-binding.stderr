error: Two subexpressions have incomparable clocks: 'when b2 and 'when b1 are incompatible
  --> tests/compile_fail/ui/clk/bad-binding.rs:10:9
   |
10 |         _n = 1 when b2;
   |         ^^^^^^^^^^^^^^
   |
note: This is clocked by 'when b1
  --> tests/compile_fail/ui/clk/bad-binding.rs:10:9
   |
10 |         _n = 1 when b2;
   |         ^^
note: defined here
  --> tests/compile_fail/ui/clk/bad-binding.rs:6:23
   |
6  |         _n : int when b1;
   |                       ^^
note: This is clocked by 'when b2
  --> tests/compile_fail/ui/clk/bad-binding.rs:10:14
   |
10 |         _n = 1 when b2;
   |              ^^^^^^^^^
note: defined here
  --> tests/compile_fail/ui/clk/bad-binding.rs:10:21
   |
10 |         _n = 1 when b2;
   |                     ^^
